pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -26,6 +26,7 @@ import (
 
 	"github.com/yl2chen/cidranger"
 	v1 "k8s.io/api/core/v1"
+	discoveryv1alpha1 "k8s.io/api/discovery/v1alpha1"
```

Comment:
it's beta in 1.17. Will the 1.17 API server send both? I guess it would have to for backward compat. It's worth noting that the structure changed a bit in 1.17.

Reply:
1.8 is released， it is beta now. but still have risk, so disabled by default.

Reply:
I'll update to beta. will do it in a separate PR (that merges before this) since this PR is big enough

Reply:
Needs some changes to operator as well, so will definitely do in another PR